File: 6KYU/Tribonacci Sequence/solution.cs
using System.Collections.Generic;
using System.Linq;
public class Xbonacci
{
  public double[] Tribonacci(double[] signature, int n)
  {
      var res = new List<double> {signature[0], signature[1], signature[2]};
      int a = 0, b = 1, c = 2;
      for(int i = 0; i < n-signature.Length; i++)
      {
          res.Add(res[a] + res[b] + res[c]);
          a++; b++; c++;
      }
      return res.Take(n).ToArray();
  }
}

File: 6KYU/Tribonacci Sequence/solution.go
package kata

func Tribonacci(signature [3]float64, n int) []float64 {
    var res []float64
    res = append(res, signature[0])
    res = append(res, signature[1])
  
    if n < 3 {
      return res[:n]      
    }
    
    a, b, c := signature[0], signature[1], signature[2]
    for i:=0; i < n-2; i++ {
      res = append(res, c)
      t := c + a + b
      a = b
      b = c
      c = t      
    }
    return res
}